internal/sql/repository/chartConfig/ConfigMapRepository.go

```diff
@@ -58,6 +60,39 @@ type ConfigMapAppModel struct {
 	SecretData    string   `sql:"secret_data"`
 	sql.AuditLog
 }
+type CMCSNames struct {
+	CMName string `json:"cm_name"`
+	CSName string `json:"cs_name"`
+}
+
+func (impl ConfigMapRepositoryImpl) GetConfigNamesAppLevel(appId int) ([]CMCSNames, error) {
+	var cMCSNames []CMCSNames
+	query := "SELECT json_array_elements(config_map_data::json->'maps')->>'name' AS cm_name,    " +
+		"json_array_elements(secret_data::json->'secrets')->>'name' AS cs_name " +
+		"FROM config_map_app_level WHERE app_id = ?;"
+
+	_, err := impl.dbConnection.Query(&cMCSNames, query, appId)
+	if err != nil && err != pg.ErrNoRows {
+		impl.Logger.Errorw("error occurred while fetching CM/CS names ", "appId", appId, "err", err)
+	} else {
+		err = nil //ignoring noRows Error
+	}
+	return cMCSNames, err
+}
+func (impl ConfigMapRepositoryImpl) GetConfigNamesEnvLevel(appId int, envId int) ([]CMCSNames, error) {
+	var cMCSNames []CMCSNames
+	query := "SELECT json_array_elements(config_map_data::json->'maps')->>'name' AS cm_name,    " +
+		"json_array_elements(secret_data::json->'secrets')->>'name' AS cs_name " +
```

Comment:
once you break this expression as per go-pg, reuse this expression in both queries

Reply:
also you can give a thought on parameterizing the repo method such that you can  have just one repo method instead of two